Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryTransforms.java

```diff
@@ -0,0 +1,187 @@
+            participantObject.setParticipantObjectQuery(byteArray);
+        }
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+        return auditMsg;
+
+    }
+
+    private byte[] getParticipantObjectQuery(PRPAIN201305UV02 request, PRPAIN201306UV02 response) throws JAXBException {
+        JAXBContextHandler oHandler = new JAXBContextHandler();
+        String JAXB_HL7_CONTEXT_NAME = "org.hl7.v3";
+        JAXBContext jc = oHandler.getJAXBContext(JAXB_HL7_CONTEXT_NAME);
+        Marshaller marshaller = jc.createMarshaller();
+        ByteArrayOutputStream baOutStrm = new ByteArrayOutputStream();
+        baOutStrm.reset();
+        if (request != null && request.getControlActProcess() != null && request.getControlActProcess().getQueryByParameter() != null) {
+            marshaller.marshal(request.getControlActProcess().getQueryByParameter(), baOutStrm);
+        } else if (response != null && response.getControlActProcess() != null && response.getControlActProcess().getQueryByParameter() != null) {
+            marshaller.marshal(response.getControlActProcess().getQueryByParameter(), baOutStrm);
+        }
+
+        return baOutStrm.toByteArray();
+    }
+
+    private II getPatientIdFromRequest(PRPAIN201305UV02 msg) {
+        PRPAIN201305UV02 oPatientDiscoveryRequestMessage = msg;
+        II oII = null;
+
+        if (oPatientDiscoveryRequestMessage != null && oPatientDiscoveryRequestMessage.getControlActProcess() != null
+            && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter() != null
+            && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue() != null
+            && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList() != null
+            && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList().getLivingSubjectId() != null
+            && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList().getLivingSubjectId().get(0) != null && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList().getLivingSubjectId().get(0).getValue() != null && oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList().getLivingSubjectId().get(0).getValue().get(0) != null) {
+            oII = oPatientDiscoveryRequestMessage.getControlActProcess().getQueryByParameter().getValue().getParameterList().getLivingSubjectId().get(0).getValue().get(0);
+
+        } else {
+            LOG.error("PatientId doesn't exists in the received PRPAIN201305UV02 message");
+            return oII;
+        }
+        return oII;
+    }
+
+    private List<II> getPatientIdFromResponse(PRPAIN201306UV02 msg) {
+        List<II> oIIs = null;
+        PRPAIN201306UV02 oPatientDiscoveryResponseMessage = msg;
```

Comment:
not sure why this line is needed, why are you assigning the variable passed in to a new variable?